recipes/bx/all/conanfile.py

```diff
@@ -1,81 +1,252 @@
+            self.binExt = ""
+            self.packageLibPrefix = "lib"
+            self.binFolder = "darwin"
+        
+        self.genieExtra = ""
+        self.projs = ["bx"]
+        if self.options.tools:
+            self.projs.extend(["bin2c", "lemon"])
+        if is_msvc(self):
+            # TODO Gotta support old "Visual Studio" compiler until 2.0, then we can remove it
+            if self.settings.compiler == "Visual Studio":
+                if self.settings.compiler.runtime in ["MD", "MDd"]:
+                    self.genieExtra += " --with-dynamic-runtime"
+            elif self.settings.compiler == "msvc":
+                if self.settings.compiler.runtime == "dynamic":
+                    self.genieExtra += " --with-dynamic-runtime"
+
+    def layout(self):
+        basic_layout(self, src_folder=".")
+
+    def validate(self):
+        if not self.settings.os == "Windows":
+            if not self.options.fPIC:
+                raise ConanInvalidConfiguration("This package does not support builds without fPIC.")
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, 14)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            try:
+                minimum_required_compiler_version = self._compiler_required[str(self.settings.compiler)]
+                if Version(self.settings.compiler.version) < minimum_required_compiler_version:
+                    raise ConanInvalidConfiguration("This package requires C++14 support. The current compiler does not support it.")
+            except KeyError:
+                self.output.warn("This recipe has no checking for the current compiler. Please consider adding it.")
+
+    def build_requirements(self):
+        self.tool_requires("genie/1170")
+        if not is_msvc(self) and self._settings_build.os == "Windows":
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                self.tool_requires("msys2/cci.latest")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = glob.glob('bx-*/')[0]
-        os.rename(extracted_dir, self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True,
+                    destination=os.path.join(self.source_folder, self._bx_folder))
 
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+    def generate(self):
+        vbe = VirtualBuildEnv(self)
+        vbe.generate()
+        if is_msvc(self):
+            tc = VCVars(self)
+            tc.generate()
+        else:
+            tc = AutotoolsToolchain(self)
+            tc.generate()
 
     def build(self):
-        cmake = self._configure_cmake()
-        cmake.build()
+        # Save bx path for convencience
+        self.bxPath = os.path.join(self.source_folder, self._bx_folder)
```

Comment:
Python/Conan uses `snake_case` so it would be nice to have it consistent my brain is freaking out with only 1 cup of coffee 🙃 just a superficial gripe not a blocker

Reply:
I'm used to default to camelCase (I just don't really like typing _ to be honest!) but I'll try to change stuff over to snake_case for consistency 😅